### PR TITLE
Fixed bug where for old periods user age was incorrectly calculated.

### LINKED ISSUE #11: [analytics] We incorrectly determine age for old users
When calculating age for users from previuos periods (week/month/...), we calc their age at the present moment, but instead we should be calculating their age as it was at that period of time (I guess at the end of it?).

So let's say I am doing report for last 4 weeks. When calculating the age of unique users in week num 2, I should calc their age by doing the difference between the last event in week num 2 and their first event they ever had, not between the last even they ever had and first event they ever had.

### PR DESCRIPTION
**Pull request by Martinsos**
Fixes #11 .

### CODE REVIEW

File: src/analytics/analytics.js [hunk 1]
```diff
@@ -338,8 +338,9 @@ function calcUniqueLocalEventsPerPeriodByAge (userEvents, periods) {
     periodEnds: [] // [string] where strings are dates formatted as YY-MM-DD.
   }
   for (let period of periods) {
-    const ids = uniqueUserIdsInPeriod(userEvents, period)
-    const ages = ids.map(id => (calcUserAgeInDays(userEvents, id)))
+    const userEventsUpToAndInPeriod = filterEventsUpToAndInPeriod(userEvents, period)
+    const ids = uniqueUserIdsInPeriod(userEventsUpToAndInPeriod, period)
+    const ages = ids.map(id => (calcUserAgeInDays(userEventsUpToAndInPeriod, id)))
```
Comment (infomiho) [inline src/analytics/analytics.js:343]:
```suggestion
    const ages = ids.map(id => calcUserAgeInDays(userEventsUpToAndInPeriod, id))
```

File: src/analytics/analytics.js [hunk 2]
```diff
@@ -375,6 +376,12 @@ function filterEventsInPeriod (es, [startTime, endTime]) {
   return es.filter(e => moment(e.timestamp).isSameOrBefore(endTime) && moment(e.timestamp).isAfter(startTime))
 }
 
+// Takes a bunch of events that have .timestamp field and returns only those that
+// happened in the period specified via (startTime, endTime) or before it.
+function filterEventsUpToAndInPeriod (es, [startTime, endTime]) {
```
Comment (infomiho) [inline src/analytics/analytics.js:381]: While we know we are talking about events here, `es` feels cryptic as a variable name. 

```suggestion
function filterEventsUpToAndInPeriod (events, [startTime, endTime]) {
```

**Review by infomiho — APPROVED** [review]
Looks good, some style changes suggested

File: src/analytics/analytics.js (hunk 2 above)
Comment (Martinsos) [reply src/analytics/analytics.js]: Yeah, this was a bit too short :D. Fixed!

**Comment by Martinsos** [comment]
Thanks, I did the suggested changes and merged!

**Pull request merged**